commit ee53a887bead08ec1354de3e74659da537f87515
Author: Simo Sorce <simo@redhat.com>
Date:   Sat Jul 20 13:19:19 2013 -0400

    Load cred store functions from GSS modules
    
    When the credential store feature was implement the related functions
    were added to struct gss_config, but the initialization function that
    dynamically loads modules was not changed to see if the plugin being
    loaded provided such functions.
    
    This will allow non-builtin mechanism and interposer mechanism to
    implement custom credential store extensions if they wish.
    
    ticket: 7682

diff --git a/src/lib/gssapi/mechglue/g_initialize.c b/src/lib/gssapi/mechglue/g_initialize.c
index f5b8b15..70cc4ee 100644
--- a/src/lib/gssapi/mechglue/g_initialize.c
+++ b/src/lib/gssapi/mechglue/g_initialize.c
@@ -680,6 +680,8 @@ build_dynamicMech(void *dl, const gss_OID mech_type)
         GSS_ADD_DYNAMIC_METHOD_NOLOOP(dl, mech, gss_inquire_mech_for_saslname);
         /* RFC 5587 */
         GSS_ADD_DYNAMIC_METHOD_NOLOOP(dl, mech, gss_inquire_attrs_for_mech);
+	GSS_ADD_DYNAMIC_METHOD(dl, mech, gss_acquire_cred_from);
+	GSS_ADD_DYNAMIC_METHOD(dl, mech, gss_store_cred_into);
 	GSS_ADD_DYNAMIC_METHOD(dl, mech, gssspi_acquire_cred_with_password);
 	GSS_ADD_DYNAMIC_METHOD(dl, mech, gssspi_import_sec_context_by_mech);
 	GSS_ADD_DYNAMIC_METHOD(dl, mech, gssspi_import_name_by_mech);
@@ -778,6 +780,8 @@ build_interMech(void *dl, const gss_OID mech_type)
 	RESOLVE_GSSI_SYMBOL(dl, mech, gss, _inquire_mech_for_saslname);
 	/* RFC 5587 */
 	RESOLVE_GSSI_SYMBOL(dl, mech, gss, _inquire_attrs_for_mech);
+	RESOLVE_GSSI_SYMBOL(dl, mech, gss, _acquire_cred_from);
+	RESOLVE_GSSI_SYMBOL(dl, mech, gss, _store_cred_into);
 	RESOLVE_GSSI_SYMBOL(dl, mech, gssspi, _acquire_cred_with_password);
 	RESOLVE_GSSI_SYMBOL(dl, mech, gssspi, _import_sec_context_by_mech);
 	RESOLVE_GSSI_SYMBOL(dl, mech, gssspi, _import_name_by_mech);
